nucypher/cli/worklock.py

```diff
@@ -0,0 +1,132 @@
+"""
+This file is part of nucypher.
+
+nucypher is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+nucypher is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with nucypher.  If not, see <https://www.gnu.org/licenses/>.
+
+"""
+
+import click
+from web3 import Web3
+
+from nucypher.blockchain.eth.agents import ContractAgency, WorkLockAgent
+from nucypher.blockchain.eth.interfaces import BlockchainInterfaceFactory
+from nucypher.blockchain.eth.registry import InMemoryContractRegistry, LocalContractRegistry
+from nucypher.characters.banners import WORKLOCK_BANNER
+from nucypher.cli.actions import select_client_account
+from nucypher.cli.config import nucypher_click_config
+from nucypher.cli.painting import paint_receipt_summary, paint_worklock_status, paint_worklock_participant_notice
+from nucypher.cli.types import EIP55_CHECKSUM_ADDRESS, WEI, EXISTING_READABLE_FILE
+
+
+@click.command()
+@click.argument('action')
+@click.option('--force', help="Don't ask for confirmation", is_flag=True)
+@click.option('--sync/--no-sync', default=False)
+@click.option('--poa', help="Inject POA middleware", is_flag=True, default=False)
+@click.option('--provider', 'provider_uri', help="Blockchain provider's URI", type=click.STRING, default="auto://")
+@click.option('--registry-filepath', help="Custom contract registry filepath", type=EXISTING_READABLE_FILE)
+@click.option('--bidder-address', help="Bidding address.", type=EIP55_CHECKSUM_ADDRESS)
+@click.option('--value', help="Bid amount in wei", type=WEI)
+@nucypher_click_config
+def worklock(click_config, action, force, provider_uri, sync, registry_filepath, poa, bidder_address, value):
+    """Participate in NU token worklock bidding."""
```

Comment:
Should we add help information for the various actions here
```
    """
    Participate in NU token worklock bidding.

    \b
    Actions
    -------------------------------------------------
    ...

    """
```

Reply:
Yes - that's definitely missing.